components/application-broker/internal/broker/provision.go

```diff
@@ -74,14 +75,19 @@ func (svc *ProvisionService) Provision(ctx context.Context, osbCtx osbContext, r
 	defer svc.mu.Unlock()
 
 	iID := internal.InstanceID(req.InstanceID)
-	paramHash := jsonhash.HashS(req.Parameters)
+	provisioningParams := &internal.ProvisioningParameters{
+		Parameters: req.Parameters,
+	}
 
 	switch state, err := svc.instanceStateGetter.IsProvisioned(iID); true {
 	case err != nil:
 		return nil, &osb.HTTPStatusCodeError{StatusCode: http.StatusBadRequest, ErrorMessage: strPtr(fmt.Sprintf("while checking if instance is already provisioned: %v", err))}
 	case state:
-		if err := svc.compareProvisioningParameters(iID, paramHash); err != nil {
-			return nil, &osb.HTTPStatusCodeError{StatusCode: http.StatusConflict, ErrorMessage: strPtr(fmt.Sprintf("while comparing provisioning parameters %v: %v", req.Parameters, err))}
+		if err := svc.compareProvisioningParameters(iID, provisioningParams); err != nil {
+			if strings.Contains(err.Error(), "conflict") { // TODO: implement custom errors for database access problems
```

Comment:
Do not analyze the error message here! You can introduce a dedicated error or just return (bool, error) instead of (error). Such value could indicate if the parameters are equal or not (and consider change of method name).